### PR TITLE
Nc news 7

### PR DESCRIPTION
**Pull request by AminahSaleem**

### CODE REVIEW

**Review by itsaliabdallah — CHANGES_REQUESTED** [review]
I'd like to see some more testing for this endpoints.
What if I send an object with more properties than I am expecting? as long as the body and username is there I would want to see a 201 success.

What if the ID is invalid?
What if the ID is valid but doesn't exist?
What if the username you try doesn't exist.

Just a few more things which I think need to be tested.

These errors will involve getting some PSQL errors back which you can add to your error handlers so send messaging back to the client.

Good work so far!

File: __tests__/api.test.js [hunk 1]
```diff
@@ -157,4 +157,68 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+                const {comment} = body
+                expect(comment).toMatchObject({
+                author: "butter_bridge",
+                body: 'Test comment'
+                })
+            })
+        })
+        test('POST 400: should respond with an error if the username does not exist', () => {
+            const newComment = {
+                username: 'Aminah',
```
Comment (poppymcd) [inline __tests__/api.test.js:204]: If the username is valid but doesn't exist, should it be a 400?

File: __tests__/api.test.js [hunk 2]
```diff
@@ -157,4 +157,68 @@ describe('GET /api/articles/:article_id/comments', () => {
             })
         })
         })
- 
\ No newline at end of file
+ 
+describe('POST /api/articles/:articles_id/comments', () => {
```
Comment (poppymcd) [inline __tests__/api.test.js:161]: As Ali said, what if we tried to post to articles/not-an-id ? It'd be great to see a test for that

**Review by poppymcd — CHANGES_REQUESTED** [review]
Nearly there! Consider the feedback I wrote in your test file and resubmit a pr tomorrow :)

**Review by itsaliabdallah — APPROVED** [review]
Nice great job actioning the feedback I have you last time!
Merge when ready

**Pull request merged**